### ISSUE TITLE
Unzip Errors

### ISSUE THREAD

**Issue by Zornhut** [jjoshpoland/MultiplePerspectiveCharacterController#1]
Hi Josh! 

This looks like a great controller set up! I downloaded it and tried unzipping the file but I'm getting an error saying that several "PropertyProvider_Cinemachine_CinemachineBrain...." files are too long.

 I tried again using Winzip and I get a different error pointing to the same files saying "Error: The system cannot find the path specified. Cannot create "C:\....PropertyProvider_Cinemachine_CinemachineBrain...." 

All the other files seem to download fine but I'm missing at least half a dozen or more files from the "Property Providers" folder. Is there a work around I can do on my end or is the original file corrupted?

Btw, sorry to post this here, I couldn't figure out a way to email you.

-Z

**Comment by jjoshpoland** [comment]
Hey there, I’m away from the desk this week, but I will take a look when I’m back. Sounds like I may need to re upload.

**Comment by jjoshpoland** [comment]
Also, have you tried cloning the repo using git/github desktop?

**Comment by Zornhut** [comment]
Thanks for the replies! 

I'm band new to git hub so I'm unfamiliar with cloning repositories. Was downloading the zip file the wrong approach to take? I actually created my account just so that I could look at this controller. I'm currently learning visual scripting and I'm having trouble finding good examples of 3rd person controllers. I was hoping to study this controller to gain a better understanding of the system. Sorry I'm a bit of a noob but I really appreciate you sharing your knowledge with the community!

**Comment by jjoshpoland** [comment]
No problem, if you get Github for Desktop, there is an option to clone a repository. You just give it a url to this repository and it will pull it down into a directory of your choosing. I'm not sure what's wrong with the zip file, but cloning it should work a lot better.

**Comment by Zornhut** [comment]
Thanks- I'll give that a try!

**Comment by Zornhut** [comment]
I made a clone and was able to import the files into unity but I'm still getting an error from the Property Providers in unity:

Assets\Unity.VisualScripting.Generated\VisualScripting.Core\Property Providers\PropertyProvider_TMPro_TextAlignmentOptions.cs(15,3): error CS0579: Duplicate 'global::Unity.VisualScripting.SerializedPropertyProviderAttribute' attribute

**Comment by jjoshpoland** [comment]
When does the error show up? Can you still run the sample scene? I don't think I use TMPro so I don't know why it would throw an error.

I was able to open the project up just now and didn't get that error.

**Comment by Zornhut** [comment]
No- I get the error when I try to run the sample scene. It's a new project and the graphs have their values so I think I imported it properly but I'm not 100% sure. 

On second review, I don't think I was pulling from source only... I'll try deleting and re-importing with source only this time.

**Comment by jjoshpoland** [comment]
The whole repo is a project, so you should be able to go in and add it via Unity Hub rather than importing. That way all the visual scripting project settings are preserved.

You can also import just the assets folder into a fresh project if you prefer and get them to work together.

**Comment by Zornhut** [comment]
Oh- gotcha! I'll try that approach. Just seeing your graphs has been really helpful, thanks for all the help! Do you happen to know any other good resources for Visual Scripting? I'm pretty new to unity and coding but I've worked in film and games for 15 years as a concept artist and production designer. I love my craft but I'd really like to make my own game for a change, haha

**Comment by jjoshpoland** [comment]
Glad to help. Unfortunately I don’t know of a lot of resources on this topic outside of learning Unreal. One of the reasons I published this was because the space is so empty and being a Unity programmer, it was easy for me to pick up.
